File: codingame/shadows_of_the_knight_ep_1/shadows_of_the_knight_ep_1.js

var inputs = readline().split(' ');
const W = parseInt(inputs[0]); // width of the building.
const H = parseInt(inputs[1]); // height of the building.
const N = parseInt(readline()); // maximum number of turns before game over.
var inputs = readline().split(' ');
var x0 = parseInt(inputs[0]);
var y0 = parseInt(inputs[1]);

var lowestY = 0;
var lowestX = 0;
var highestY = H;
var highestX = W;

// game loop
while (true) {
    const bombDir = readline(); // the direction of the bombs from batman's current location (U, UR, R, DR, D, DL, L or UL)
    
    switch(bombDir[0]) {
        case 'D': lowestY = y0 + 1; break;
        case 'U': highestY = y0 - 1; break;
        case 'L': highestX = x0 - 1; break;
        case 'R': lowestX = x0 + 1; break;
    }
    switch(bombDir[1]) {
        case 'L': highestX = x0 - 1; break;
        case 'R': lowestX = x0 + 1; break;
    }

    x0 = Math.floor((highestX + lowestX) / 2);
    y0 = Math.floor((highestY + lowestY) / 2);

    console.log(x0, y0);

}

File: codingame/shadows_of_the_knight_ep_1/shadows_of_the_knight_ep_1.py
#!/usr/bin/python

# width and height of the building
width, height = [int(i) for i in input().split()]
turns = int(input())  # maximum number of turns before game over.
batman_x, batman_y = [int(i) for i in input().split()]
lowest_x, lowest_y, highest_x, highest_y = [0, 0, width, height]

# game loop
while True:
    bomb_dir = input()  # the direction of the bombs from batman's current location (U, UR, R, DR, D, DL, L or UL)

    if "D" in bomb_dir:
        lowest_y = batman_y + 1

    elif "U" in bomb_dir:
        highest_y = batman_y - 1
    
    if "L" in bomb_dir:
        highest_x = batman_x - 1
    
    elif "R" in bomb_dir:
        lowest_x = batman_x + 1

    batman_y = (highest_y + lowest_y) // 2
    batman_x = (highest_x + lowest_x) // 2

    # the location of the next window Batman should jump to.
    print(batman_x, batman_y)
    
